x-pack/plugins/lens/public/indexpattern_datasource/index.ts

```diff
@@ -35,7 +41,12 @@ export class IndexPatternDatasource {
 
   setup(
     core: CoreSetup<IndexPatternDatasourceStartPlugins>,
-    { expressions, editorFrame, charts, data: dataSetup }: IndexPatternDatasourceSetupPlugins
+    {
+      fieldFormats: fieldFormatsSetup,
+      expressions,
+      editorFrame,
+      charts,
+    }: IndexPatternDatasourceSetupPlugins
   ) {
     editorFrame.registerDatasource(async () => {
       const {
```

Comment:
registration of functions on the client seems to happen late (after setup lifecycle)

this should be fixed (not necesarry as part of this PR) as we plan to introduce checks in expression service that will prevent this.

Reply:
yes you are right, I also wanted to highlight that in my PR, but then decided that it's not related....  In Lens we have some setup methods which executed on start phase =(

Reply:
Yes this is important, but we will take care of it in another PR